test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/FreezeUnfreezeTokenPrecompileSuite.java

```diff
@@ -95,7 +93,136 @@ public List<HapiApiSpec> getSpecsInSuite() {
         return List.of(
                 freezeUnfreezeFungibleWithNegativeCases(),
                 freezeUnfreezeNftsWithNegativeCases(),
-                isFrozenHappyPathWithLocalCall());
+                isFrozenHappyPathWithLocalCall(),
+                noTokenIdReverts(),
+                noKeyReverts());
+    }
+
+    private HapiApiSpec noTokenIdReverts() {
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
```

Comment:
nit: vanillaTokenID can be an instance variable instead since it's used in every test case.